# **ANALYSE ET TRAITEMENT ALL DATA EHCVM - ANSTAT**

# **MISE EN SITUATION**

# **RECUPERATION DES DONNEES**

In [63]:
#importation des bibliothèques

import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [64]:
# Load data

df = pd.read_csv("../DATAS/ANSTAT2021_dataset_AllVars.csv")

# **DESCRIPTION ET OBJECTIFS**

# **ANALYSE EXPLORATOIRE DES DONNEES**

In [65]:
# Lecture du dataframe

df.head()

idmen  idind country  year       PanelHH  numind  vague agro_zone   zaemil  \
0      2      4     CIV  2021  Menage panel       4      1   ABIDJAN  ABIDJAN   
1      2      5     CIV  2021  Menage panel       5      1   ABIDJAN  ABIDJAN   
2      2      3     CIV  2021  Menage panel       3      1   ABIDJAN  ABIDJAN   
3      2      6     CIV  2021  Menage panel       6      1   ABIDJAN  ABIDJAN   
4      2      2     CIV  2021  Menage panel       2      1   ABIDJAN  ABIDJAN   

          region_name  ... elec_ua   tv  fer  frigo cuisin ordin  decod  car  \
0  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui    Non   Non    Oui  Non   
1  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui    Non   Non    Oui  Non   
2  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui    Non   Non    Oui  Non   
3  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui    Non   Non    Oui  Non   
4  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui    Non   Non    Oui  Non   

  superf bancarise  
0    0.0         1  
1    0.0         0  
2    0.0         0  
3    0.0         0  
4    0.0         0  

[5 rows x 78 columns]

In [66]:
# Copy du dataframe afin d'avoir toujours une version originale

df_anstat = df.copy()
df_anstat

idmen  idind country  year       PanelHH  numind  vague agro_zone  \
0          2      4     CIV  2021  Menage panel       4      1   ABIDJAN   
1          2      5     CIV  2021  Menage panel       5      1   ABIDJAN   
2          2      3     CIV  2021  Menage panel       3      1   ABIDJAN   
3          2      6     CIV  2021  Menage panel       6      1   ABIDJAN   
4          2      2     CIV  2021  Menage panel       2      1   ABIDJAN   
...      ...    ...     ...   ...           ...     ...    ...       ...   
64486  13863  64490     CIV  2021  Menage panel       5      2    CENTRE   
64487  13863  64489     CIV  2021  Menage panel       4      2    CENTRE   
64488  13863  64491     CIV  2021  Menage panel       6      2    CENTRE   
64489  13863  64488     CIV  2021  Menage panel       3      2    CENTRE   
64490  13863  64486     CIV  2021  Menage panel       1      2    CENTRE   

               zaemil         region_name  ... elec_ua   tv  fer  frigo  \
0             ABIDJAN  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui   
1             ABIDJAN  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui   
2             ABIDJAN  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui   
3             ABIDJAN  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui   
4             ABIDJAN  AUTONOME D'ABIDJAN  ...     Non  Oui  Non    Oui   
...               ...                 ...  ...     ...  ...  ...    ...   
64486  CENTRE (rural)             MORONOU  ...     Non  Oui  Non    Non   
64487  CENTRE (rural)             MORONOU  ...     Non  Oui  Non    Non   
64488  CENTRE (rural)             MORONOU  ...     Non  Oui  Non    Non   
64489  CENTRE (rural)             MORONOU  ...     Non  Oui  Non    Non   
64490  CENTRE (rural)             MORONOU  ...     Non  Oui  Non    Non   

      cuisin ordin  decod  car     superf bancarise  
0        Non   Non    Oui  Non        0.0         1  
1        Non   Non    Oui  Non        0.0         0  
2        Non   Non    Oui  Non        0.0         0  
3        Non   Non    Oui  Non        0.0         0  
4        Non   Non    Oui  Non        0.0         0  
...      ...   ...    ...  ...        ...       ...  
64486    Non   Non    Oui  Non  8566398.0         0  
64487    Non   Non    Oui  Non  8566398.0         0  
64488    Non   Non    Oui  Non  8566398.0         0  
64489    Non   Non    Oui  Non  8566398.0         0  
64490    Non   Non    Oui  Non  8566398.0         1  

[64491 rows x 78 columns]

In [67]:
# Nombre lignes et colonnes

df_anstat.shape

(64491, 78)

In [68]:
# Infos dataframe

df_anstat.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64491 entries, 0 to 64490
Data columns (total 78 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   idmen             64491 non-null  int64  
 1   idind             64491 non-null  int64  
 2   country           64491 non-null  object 
 3   year              64491 non-null  int64  
 4   PanelHH           64491 non-null  object 
 5   numind            64491 non-null  int64  
 6   vague             64491 non-null  int64  
 7   agro_zone         64491 non-null  object 
 8   zaemil            64491 non-null  object 
 9   region_name       64491 non-null  object 
 10  department        64491 non-null  object 
 11  city              64491 non-null  object 
 12  milieu_resid      64491 non-null  object 
 13  hhweight          64491 non-null  float64
 14  resid             64491 non-null  object 
 15  sex               64491 non-null  object 
 16  age_num           64491 non-null  int64 

In [69]:
# Description du Dataframe

df_anstat.describe()

idmen         idind     year        numind         vague  \
count  64491.000000  64491.000000  64491.0  64491.000000  64491.000000   
mean    7086.913678  32246.000000   2021.0      3.794529      1.488719   
std     3947.746739  18617.092442      0.0      2.630311      0.499877   
min        2.000000      1.000000   2021.0      1.000000      1.000000   
25%     3727.000000  16123.500000   2021.0      2.000000      1.000000   
50%     7173.000000  32246.000000   2021.0      3.000000      1.000000   
75%    10389.000000  48368.500000   2021.0      5.000000      2.000000   
max    13863.000000  64491.000000   2021.0     28.000000      2.000000   

           hhweight      age_num        agemar        volhor       salaire  \
count  64491.000000  64491.00000  23276.000000  18694.000000  6.449100e+04   
mean     466.540740     23.28083     22.838847   1833.543949  1.012867e+05   
std      509.031829     19.32949      6.171423    872.591731  5.634367e+05   
min       40.775154      0.00000     10.000000      6.000000  0.000000e+00   
25%      217.131060      8.00000     18.000000   1200.000000  0.000000e+00   
50%      345.130950     17.00000     21.000000   1920.000000  0.000000e+00   
75%      547.540710     36.00000     26.000000   2400.000000  0.000000e+00   
max    13130.647000    122.00000     70.000000   4500.000000  2.720000e+07   

       ...  salaire_mois  salaire_sec_mois  rev_total_mois    log_revenu  \
count  ...  4.770000e+03      4.600000e+02    6.449100e+04  64491.000000   
mean   ...  1.141174e+05      5.489116e+04    8.832082e+03      0.854420   
std    ...  1.332292e+05      8.480195e+04    4.837005e+04      2.993787   
min    ...  0.000000e+00      0.000000e+00    0.000000e+00      0.000000   
25%    ...  4.333333e+04      1.800000e+04    0.000000e+00      0.000000   
50%    ...  7.000000e+04      4.000000e+04    0.000000e+00      0.000000   
75%    ...  1.383559e+05      7.065000e+04    0.000000e+00      0.000000   
max    ...  2.266666e+06      1.500001e+06    2.522333e+06     14.740695   

        sans_revenu   a_assurance     alphabete  est_scolarise        superf  \
count  64491.000000  64491.000000  64491.000000   64491.000000  6.449100e+04   
mean       0.924253      0.034858      0.448373       0.265029  5.443669e+07   
std        0.264595      0.183420      0.497331       0.441352  2.361731e+09   
min        0.000000      0.000000      0.000000       0.000000  0.000000e+00   
25%        1.000000      0.000000      0.000000       0.000000  0.000000e+00   
50%        1.000000      0.000000      0.000000       0.000000  1.139040e+00   
75%        1.000000      0.000000      1.000000       1.000000  8.750000e+00   
max        1.000000      1.000000      1.000000       1.000000  1.614127e+11   

          bancarise  
count  64491.000000  
mean       0.267464  
std        0.442640  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000  

[8 rows x 22 columns]

In [70]:
# Valeurs manquantes

df_anstat.isna().sum()

idmen        0
idind        0
country      0
year         0
PanelHH      0
            ..
ordin        0
decod        0
car          0
superf       0
bancarise    0
Length: 78, dtype: int64

In [71]:
missing_counts = df_anstat.isnull().sum()
missing_counts = missing_counts[missing_counts > 0]

print("Variables avec valeurs manquantes :")
print(missing_counts)

Variables avec valeurs manquantes :
ethnie              12150
agemar              41215
aff30j              47271
durarr              53284
con30j              47271
handit               9567
handig               9567
educ_scol           47400
educ_hi                 4
branch              38197
sectins             38197
csp                 38197
volhor              45797
sectins_sec         61126
csp_sec             61126
volhor_sec          61126
serviceconsult      47271
persconsult         47271
salaire_mois        59721
salaire_sec_mois    64031
dtype: int64


In [72]:
# Nombre de valeurs uniques

df_anstat.nunique()

idmen        12965
idind        64491
country          1
year             1
PanelHH          2
             ...  
ordin            2
decod            2
car              2
superf        6163
bancarise        2
Length: 78, dtype: int64

In [73]:
# Listing des colonnes du dataframe

df_anstat.columns

Index(['idmen', 'idind', 'country', 'year', 'PanelHH', 'numind', 'vague',
       'agro_zone', 'zaemil', 'region_name', 'department', 'city',
       'milieu_resid', 'hhweight', 'resid', 'sex', 'age_num', 'lien',
       'marital_status', 'religion', 'ethnie', 'nation', 'agemar', 'mal30j',
       'aff30j', 'arrmal', 'durarr', 'con30j', 'hos12m', 'couvmal', 'moustiq',
       'handit', 'handig', 'alfa', 'alfa2', 'scol', 'educ_scol', 'educ_hi',
       'diplome', 'telpor', 'internet', 'activ7j', 'activ12m', 'branch',
       'sectins', 'csp', 'volhor', 'salaire', 'emploi_sec', 'sectins_sec',
       'csp_sec', 'volhor_sec', 'salaire_sec', 'has_account_bank',
       'serviceconsult', 'persconsult', 'salaire_mois', 'salaire_sec_mois',
       'rev_total_mois', 'log_revenu', 'sans_revenu', 'age_grp', 'a_assurance',
       'alphabete', 'est_scolarise', 'logem', 'elec_ac', 'elec_ur', 'elec_ua',
       'tv', 'fer', 'frigo', 'cuisin', 'ordin', 'decod', 'car', 'superf',
       'bancarise'],
      dtype=

# **NETTOYAGE DES DONNEES**

## **Traitement des valeurs manquantes** 

In [74]:
missing_counts = df_anstat.isnull().sum()
missing_counts = missing_counts[missing_counts > 0]

print("Variables avec valeurs manquantes :")
print(missing_counts)

Variables avec valeurs manquantes :
ethnie              12150
agemar              41215
aff30j              47271
durarr              53284
con30j              47271
handit               9567
handig               9567
educ_scol           47400
educ_hi                 4
branch              38197
sectins             38197
csp                 38197
volhor              45797
sectins_sec         61126
csp_sec             61126
volhor_sec          61126
serviceconsult      47271
persconsult         47271
salaire_mois        59721
salaire_sec_mois    64031
dtype: int64


### Handicap

In [75]:
df_han = df_anstat[["handit", "handig"]]
df_han = df_han[df_han["handit"].isnull()]
df_han

handit handig
6        NaN    NaN
7        NaN    NaN
18       NaN    NaN
37       NaN    NaN
41       NaN    NaN
...      ...    ...
64460    NaN    NaN
64461    NaN    NaN
64463    NaN    NaN
64481    NaN    NaN
64489    NaN    NaN

[9567 rows x 2 columns]

In [76]:
df_anstat[["handit", "handig"]] = df_anstat[["handit", "handig"]].fillna('Non')
df_han = df_anstat[["handit", "handig"]]
#df_han = df_han[df_han["handit"].isnull()]
df_han.isna().sum()

handit    0
handig    0
dtype: int64

### Ethnie / Age du 1er mariage

In [77]:
df_anstat['agemar'] = df_anstat['agemar'].fillna(-1)
df_anstat['ethnie'] = df_anstat['ethnie'].fillna('Inconnue')

### Education

In [78]:
df_edu = df_anstat[["educ_scol", "educ_hi"]]
df_edu = df_edu[df_edu["educ_hi"].isnull()]

#df_edu.isna().sum()
df_edu

educ_scol educ_hi
31672       NaN     NaN
42674       NaN     NaN
51712       NaN     NaN
61264       NaN     NaN

In [79]:
df_anstat[['educ_scol', 'educ_hi']] = df_anstat[['educ_scol', 'educ_hi']].fillna('Aucun')
df_edu = df_anstat[["educ_scol", "educ_hi"]]
df_edu

educ_scol       educ_hi
0                                       Aucun  Second. gl 2
1                       Secondaire 2 générale  Second. gl 2
2      Secondaire 1  (Post Primaire) générale  Second. gl 1
3                                    Primaire      Primaire
4                                       Aucun         Aucun
...                                       ...           ...
64486                                Primaire      Primaire
64487                                Primaire      Primaire
64488                                Primaire      Primaire
64489                                   Aucun         Aucun
64490                                   Aucun      Primaire

[64491 rows x 2 columns]

In [80]:
df_edu.isna().sum()

educ_scol    0
educ_hi      0
dtype: int64

### Vie Pro

#### Emploi secondaire

In [81]:
df_anstat["emploi_sec"]

0        Non
1        Non
2        Non
3        Non
4        Non
        ... 
64486    Non
64487    Non
64488    Non
64489    Non
64490    Oui
Name: emploi_sec, Length: 64491, dtype: object

In [82]:
df_emploi = df_anstat[["emploi_sec", "volhor_sec", "sectins_sec", "csp_sec", "salaire_sec_mois"]]
df_emploi

emploi_sec  volhor_sec        sectins_sec  \
0            Non         NaN                NaN   
1            Non         NaN                NaN   
2            Non         NaN                NaN   
3            Non         NaN                NaN   
4            Non         NaN                NaN   
...          ...         ...                ...   
64486        Non         NaN                NaN   
64487        Non         NaN                NaN   
64488        Non         NaN                NaN   
64489        Non         NaN                NaN   
64490        Oui      1152.0  Entreprise privée   

                                                 csp_sec  salaire_sec_mois  
0                                                    NaN               NaN  
1                                                    NaN               NaN  
2                                                    NaN               NaN  
3                                                    NaN               NaN  
4                                                    NaN               NaN  
...                                                  ...               ...  
64486                                                NaN               NaN  
64487                                                NaN               NaN  
64488                                                NaN               NaN  
64489                                                NaN               NaN  
64490  Travailleur familial contribuant à une entrepr...               NaN  

[64491 rows x 5 columns]

##### N'a pas de travail secondaire

In [83]:
mask = df_anstat["emploi_sec"] == "Non"

In [84]:
df_anstat.loc[mask, "volhor_sec"] = df_anstat.loc[mask, "volhor_sec"].fillna(0)
df_anstat.loc[mask, "salaire_sec_mois"] = df_anstat.loc[mask, "salaire_sec_mois"].fillna(0)

df_anstat.loc[mask, "sectins_sec"] = df_anstat.loc[mask, "sectins_sec"].fillna("Non connu")
df_anstat.loc[mask, "csp_sec"] = df_anstat.loc[mask, "csp_sec"].fillna("Non connu")

##### A un travail secondaire

In [85]:
mask2 = df_anstat["emploi_sec"] == "Oui"

In [86]:
df_anstat.loc[mask2, "salaire_sec_mois"] = df_anstat.loc[mask2, "salaire_sec_mois"].fillna(0)

In [87]:
df_emploi.isna().sum()

emploi_sec              0
volhor_sec          61126
sectins_sec         61126
csp_sec             61126
salaire_sec_mois    64031
dtype: int64

#### Volume horaire / Secteur institutionnel / Catégorie Socio-démographique / Branche d'activité

In [88]:
df_volhor = df_anstat[["volhor", "salaire", "salaire_mois", "branch", "sectins", "csp"]]
df_volhor

volhor    salaire  salaire_mois              branch  \
0      1602.1918  3350928.0      279244.0       Aut. services   
1            NaN        0.0           NaN                 NaN   
2            NaN        0.0           NaN                 NaN   
3            NaN        0.0           NaN                 NaN   
4            NaN        0.0           NaN                 NaN   
...          ...        ...           ...                 ...   
64486        NaN        0.0           NaN                 NaN   
64487        NaN        0.0           NaN                 NaN   
64488        NaN        0.0           NaN                 NaN   
64489        NaN        0.0           NaN                 NaN   
64490   864.0000        0.0           NaN  Elevage/syl./peche   

                                 sectins                             csp  
0      Entreprise publique/ parapublique   Cadre moyen/agent de maîtrise  
1                                    NaN                             NaN  
2                                    NaN                             NaN  
3                                    NaN                             NaN  
4                                    NaN                             NaN  
...                                  ...                             ...  
64486                                NaN                             NaN  
64487                                NaN                             NaN  
64488                                NaN                             NaN  
64489                                NaN                             NaN  
64490                  Entreprise Privée  Travailleur pour compte propre  

[64491 rows x 6 columns]

In [89]:
df_volhor.isna().sum()

volhor          45797
salaire             0
salaire_mois    59721
branch          38197
sectins         38197
csp             38197
dtype: int64

In [90]:
cat_cols = ['branch', 'sectins', 'csp']
for col in cat_cols:
    df_anstat[col] = df_anstat[col].fillna('Non connu')
    
    # Optionnel : imputation par mode
    # mode_val = df_anstat[col].mode()[0]
    # df_anstat[col] = df_anstat[col].fillna(mode_val)

In [91]:
num_cols = ['volhor', 'salaire_mois']
for col in num_cols:
    df_anstat[col+'_manquant'] = df_anstat[col].isnull().astype(int)  # variable indicatrice
    # Imputation
    df_anstat[col] = df_anstat[col].fillna(0)  # ou méthode médiane : df_anstat[col].fillna(df_anstat[col].median())

In [92]:
df_volhor = df_anstat[["volhor", "salaire", "salaire_mois", "branch", "sectins", "csp"]]
df_volhor.isna().sum()

volhor          0
salaire         0
salaire_mois    0
branch          0
sectins         0
csp             0
dtype: int64

### Maladie et santé 

#### Maladale les 30 derniers jours

In [93]:
df_anst = df_anstat[["mal30j", "aff30j"]]

In [94]:
df_anst = df_anst[df_anst["mal30j"] == "Non"]
df_anst

mal30j aff30j
0        Non    NaN
1        Non    NaN
2        Non    NaN
3        Non    NaN
4        Non    NaN
...      ...    ...
64480    Non    NaN
64481    Non    NaN
64482    Non    NaN
64483    Non    NaN
64484    Non    NaN

[47271 rows x 2 columns]

In [95]:
df_anst.isna().sum()

mal30j        0
aff30j    47271
dtype: int64

In [96]:
df_anstat.loc[(df_anstat['mal30j'] == 'Non') & (df_anstat['aff30j'].isnull()), 'aff30j'] = 'Aucun'

In [97]:
df_anst = df_anstat[["mal30j", "aff30j"]]
df_anst = df_anst[df_anst["mal30j"] == "Non"]
df_anst

mal30j aff30j
0        Non  Aucun
1        Non  Aucun
2        Non  Aucun
3        Non  Aucun
4        Non  Aucun
...      ...    ...
64480    Non  Aucun
64481    Non  Aucun
64482    Non  Aucun
64483    Non  Aucun
64484    Non  Aucun

[47271 rows x 2 columns]

#### Arrêt maladie

In [98]:
df_arr = df_anstat[["arrmal", "durarr"]]
df_arr = df_arr[df_arr["arrmal"] == "Non"]
df_arr

arrmal durarr
0        Non    NaN
1        Non    NaN
2        Non    NaN
3        Non    NaN
4        Non    NaN
...      ...    ...
64480    Non    NaN
64481    Non    NaN
64482    Non    NaN
64483    Non    NaN
64484    Non    NaN

[53284 rows x 2 columns]

In [99]:
df_arr.isna().sum()

arrmal        0
durarr    53284
dtype: int64

In [100]:
df_anstat.loc[(df_anstat['arrmal'] == 'Non') & (df_anstat['durarr'].isnull()), 'durarr'] = 'Aucun'
df_arr = df_anstat[["arrmal", "durarr"]]
df_arr = df_arr[df_arr["arrmal"] == "Non"]
df_arr

arrmal durarr
0        Non  Aucun
1        Non  Aucun
2        Non  Aucun
3        Non  Aucun
4        Non  Aucun
...      ...    ...
64480    Non  Aucun
64481    Non  Aucun
64482    Non  Aucun
64483    Non  Aucun
64484    Non  Aucun

[53284 rows x 2 columns]

In [101]:
df_anstat.loc[df_anstat['con30j'].isnull(), 'con30j'] = 'Non'
df_anstat['con30j']

0        Non
1        Non
2        Non
3        Non
4        Non
        ... 
64486    Oui
64487    Oui
64488    Oui
64489    Oui
64490    Oui
Name: con30j, Length: 64491, dtype: object

#### Service Santé

In [102]:
df_sant = df_anstat[["serviceconsult", "persconsult"]]
#df_sant = df_sant[df_sant["educ_hi"].isnull()]

#df_sant.isna().sum()
df_sant

serviceconsult persconsult
0                NaN         NaN
1                NaN         NaN
2                NaN         NaN
3                NaN         NaN
4                NaN         NaN
...              ...         ...
64486    Dispensaire   Infirmier
64487    Dispensaire   Infirmier
64488    Dispensaire   Infirmier
64489    Dispensaire   Infirmier
64490    Dispensaire   Infirmier

[64491 rows x 2 columns]

In [103]:
df_anstat[['serviceconsult', 'persconsult']] = df_anstat[['serviceconsult', 'persconsult']].fillna('Pas de consultation')
df_sant = df_anstat[["serviceconsult", "persconsult"]]
df_sant

serviceconsult          persconsult
0      Pas de consultation  Pas de consultation
1      Pas de consultation  Pas de consultation
2      Pas de consultation  Pas de consultation
3      Pas de consultation  Pas de consultation
4      Pas de consultation  Pas de consultation
...                    ...                  ...
64486          Dispensaire            Infirmier
64487          Dispensaire            Infirmier
64488          Dispensaire            Infirmier
64489          Dispensaire            Infirmier
64490          Dispensaire            Infirmier

[64491 rows x 2 columns]

In [104]:
df_sant.isna().sum()

serviceconsult    0
persconsult       0
dtype: int64

### Revérification

In [105]:
missing_counts = df_anstat.isnull().sum()
missing_counts = missing_counts[missing_counts > 0]

print("Variables avec valeurs manquantes :")
print(missing_counts)

Variables avec valeurs manquantes :
Series([], dtype: int64)


## **Transforme variable marié** 

In [106]:
val_mar = "Marié(e)"

df_anstat["marital_status"] = df_anstat["marital_status"].replace(["Marié(e) polygame", "Marié(e) monogame"], val_mar)

In [107]:
df_anstat["marital_status"]

0        Célibataire
1        Célibataire
2        Célibataire
3        Célibataire
4           Marié(e)
            ...     
64486    Célibataire
64487    Célibataire
64488    Célibataire
64489    Célibataire
64490       Marié(e)
Name: marital_status, Length: 64491, dtype: object

## **Nettoyage variable logement** 

In [108]:
df_anstat["logem"].unique()

array(['Locataire', 'Autre', 'Proprietaire titre',
       'Proprietaire sans titre', '9'], dtype=object)

In [109]:
df_log = df_anstat[df_anstat["logem"] == "9"]
df_log

idmen  idind country  year           PanelHH  numind  vague  agro_zone  \
755      216    760     CIV  2021  Menage non panel       5      1    ABIDJAN   
756      216    756     CIV  2021  Menage non panel       1      1    ABIDJAN   
757      216    758     CIV  2021  Menage non panel       3      1    ABIDJAN   
758      216    759     CIV  2021  Menage non panel       4      1    ABIDJAN   
759      216    757     CIV  2021  Menage non panel       2      1    ABIDJAN   
8739    1998   8740     CIV  2021      Menage panel       1      2       NORD   
8740    1998   8741     CIV  2021      Menage panel       2      2       NORD   
8765    2004   8766     CIV  2021  Menage non panel       1      2       NORD   
8766    2004   8767     CIV  2021  Menage non panel       2      2       NORD   
8767    2004   8768     CIV  2021  Menage non panel       3      2       NORD   
31985   7124  31986     CIV  2021      Menage panel       1      2  SUD-OUEST   
57852  12477  57855     CIV  2021  Menage non panel       3      1    SUD-EST   
57853  12477  57854     CIV  2021  Menage non panel       2      1    SUD-EST   
57854  12477  57856     CIV  2021  Menage non panel       4      1    SUD-EST   
57855  12477  57858     CIV  2021  Menage non panel       6      1    SUD-EST   
57856  12477  57853     CIV  2021  Menage non panel       1      1    SUD-EST   
57857  12477  57857     CIV  2021  Menage non panel       5      1    SUD-EST   

                  zaemil         region_name  ...  fer frigo cuisin  ordin  \
755              ABIDJAN  AUTONOME D'ABIDJAN  ...  Non   Oui    Oui    Non   
756              ABIDJAN  AUTONOME D'ABIDJAN  ...  Non   Oui    Oui    Non   
757              ABIDJAN  AUTONOME D'ABIDJAN  ...  Non   Oui    Oui    Non   
758              ABIDJAN  AUTONOME D'ABIDJAN  ...  Non   Oui    Oui    Non   
759              ABIDJAN  AUTONOME D'ABIDJAN  ...  Non   Oui    Oui    Non   
8739       NORD (urbain)                PORO  ...  Non   Non    Non    Non   
8740       NORD (urbain)                PORO  ...  Non   Non    Non    Non   
8765       NORD (urbain)                PORO  ...  Non   Oui    Non    Non   
8766       NORD (urbain)                PORO  ...  Non   Oui    Non    Non   
8767       NORD (urbain)                PORO  ...  Non   Oui    Non    Non   
31985  SUD-OUEST (rural)         LÔH-DJIBOUA  ...  Non   Non    Non    Non   
57852    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   
57853    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   
57854    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   
57855    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   
57856    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   
57857    SUD-EST (rural)               LA ME  ...  Non   Oui    Non    Non   

      decod  car  superf bancarise volhor_manquant salaire_mois_manquant  
755     Oui  Non     0.0         0               1                     1  
756     Oui  Non     0.0         1               0                     0  
757     Oui  Non     0.0         0               1                     1  
758     Oui  Non     0.0         0               1                     1  
759     Oui  Non     0.0         1               0                     1  
8739    Non  Non     0.0         1               0                     1  
8740    Non  Non     0.0         1               1                     1  
8765    Non  Non     0.0         1               0                     0  
8766    Non  Non     0.0         0               1                     1  
8767    Non  Non     0.0         1               1                     1  
31985   Non  Non     0.0         1               0                     0  
57852   Non  Non     0.0         0               1                     1  
57853   Non  Non     0.0         0               1                     1  
57854   Non  Non     0.0         0               1                     1  
57855   Non  Non  

In [110]:
df_log.count()

idmen                    17
idind                    17
country                  17
year                     17
PanelHH                  17
                         ..
car                      17
superf                   17
bancarise                17
volhor_manquant          17
salaire_mois_manquant    17
Length: 80, dtype: int64

In [111]:
df_anstat.count()

idmen                    64491
idind                    64491
country                  64491
year                     64491
PanelHH                  64491
                         ...  
car                      64491
superf                   64491
bancarise                64491
volhor_manquant          64491
salaire_mois_manquant    64491
Length: 80, dtype: int64

#### 17 par rapport à 64491 est négligeable.
### 64491 -> 100%
### 17 -> ?%
### 17 -> 0.03% des données
#### Ce retrouver donc avec 64474 / 64491 données cleans, fiables est acceptable.
#### Donc supprimons les lignes manquantes

In [112]:
df_anstat = df_anstat.drop(df_log.index)
df_anstat.count()

idmen                    64474
idind                    64474
country                  64474
year                     64474
PanelHH                  64474
                         ...  
car                      64474
superf                   64474
bancarise                64474
volhor_manquant          64474
salaire_mois_manquant    64474
Length: 80, dtype: int64

## **CRÉATION DE NOUVELLES VARIABLES**

In [113]:
df_anstat["sectins"].unique()

array(['Entreprise publique/ parapublique', 'Non connu',
       'Entreprise Privée', 'Etat/Collectivités locales',
       'Ménage comme employeur de personnel domestique',
       'Entreprise associative', 'Organisme international /Ambassade'],
      dtype=object)

In [114]:
df_anstat["csp"].unique()

array(['Cadre moyen/agent de maîtrise', 'Non connu',
       'Travailleur pour compte propre', 'Stagiaire ou Apprenti rénuméré',
       'Ouvrier ou employé non qualifié',
       'Stagiaire ou Apprenti non rénuméré',
       'Ouvrier ou employé qualifié',
       'Travailleur Familial contribuant pour une entreprise familial',
       'Manœuvre, aide ménagère', 'Patron/Employeur', 'Cadre supérieur'],
      dtype=object)

In [115]:
df_anstat["branch"].unique()

array(['Aut. services', 'Non connu', 'Education/Sante', 'Commerce',
       'Autr. indust.', 'Trans./Comm.', 'Restaurant/Hotel', 'BTP',
       'Elevage/syl./peche', 'Indust. extr.', 'Agriculture'], dtype=object)

In [116]:
def emp_form(row):
    # Formelle si secteur institutionnel clairement formel
    if row['sectins'] in [
        "Etat/Collectivités locales",
        "Entreprise publique/ parapublique",
        "Entreprise Privée",
        "Organisme international /Ambassade"
    ]:
        # Si le secteur institutionnel n’est pas discriminant, utiliser CSP et branche
        if row['csp'] in [
            "Cadre supérieur",
            "Cadre moyen/agent de maîtrise",
            "Ouvrier ou employé qualifié",    
            "Stagiaire ou Apprenti rénuméré",
            "Stagiaire ou Apprenti non rénuméré"
        ]:
            if row['branch'] not in ["Non_connu", "Agriculture", "Elevage/syl./peche", "Commerce"]:
                return 1
    return 0

# Application sur ton DataFrame
df_anstat['Stg_formel'] = df_anstat.apply(emp_form, axis=1)
df_anstat
#pd.crosstab(df_anstat['empl_formel'], df_anstat['Stg_formel'], normalize='index') * 100

idmen  idind country  year       PanelHH  numind  vague agro_zone  \
0          2      4     CIV  2021  Menage panel       4      1   ABIDJAN   
1          2      5     CIV  2021  Menage panel       5      1   ABIDJAN   
2          2      3     CIV  2021  Menage panel       3      1   ABIDJAN   
3          2      6     CIV  2021  Menage panel       6      1   ABIDJAN   
4          2      2     CIV  2021  Menage panel       2      1   ABIDJAN   
...      ...    ...     ...   ...           ...     ...    ...       ...   
64486  13863  64490     CIV  2021  Menage panel       5      2    CENTRE   
64487  13863  64489     CIV  2021  Menage panel       4      2    CENTRE   
64488  13863  64491     CIV  2021  Menage panel       6      2    CENTRE   
64489  13863  64488     CIV  2021  Menage panel       3      2    CENTRE   
64490  13863  64486     CIV  2021  Menage panel       1      2    CENTRE   

               zaemil         region_name  ... frigo cuisin ordin  decod  car  \
0             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
1             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
2             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
3             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
4             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
...               ...                 ...  ...   ...    ...   ...    ...  ...   
64486  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   
64487  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   
64488  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   
64489  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   
64490  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   

          superf  bancarise volhor_manquant salaire_mois_manquant Stg_formel  
0            0.0          1               0                     0          1  
1            0.0          0               1                     1          0  
2            0.0          0               1                     1          0  
3            0.0          0               1                     1          0  
4            0.0          0               1                     1          0  
...          ...        ...             ...                   ...        ...  
64486  8566398.0          0               1                     1          0  
64487  8566398.0          0               1                     1          0  
64488  8566398.0          0               1                     1          0  
64489  8566398.0          0               1                     1          0  
64490  8566398.0          1               0                     1          0  

[64474 rows x 81 columns]

In [117]:
df_anstat[df_anstat['Stg_formel'] == 1]

idmen  idind country  year       PanelHH  numind  vague agro_zone  \
0          2      4     CIV  2021  Menage panel       4      1   ABIDJAN   
40        14     36     CIV  2021  Menage panel       1      2   ABIDJAN   
44        15     42     CIV  2021  Menage panel       1      2   ABIDJAN   
46        15     43     CIV  2021  Menage panel       2      2   ABIDJAN   
49        16     49     CIV  2021  Menage panel       1      2   ABIDJAN   
...      ...    ...     ...   ...           ...     ...    ...       ...   
64240  13803  64237     CIV  2021  Menage panel       1      2    CENTRE   
64254  13807  64255     CIV  2021  Menage panel       1      2    CENTRE   
64289  13816  64285     CIV  2021  Menage panel       1      1    CENTRE   
64315  13819  64316     CIV  2021  Menage panel       2      1    CENTRE   
64316  13819  64315     CIV  2021  Menage panel       1      1    CENTRE   

               zaemil         region_name  ... frigo cuisin ordin  decod  car  \
0             ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
40            ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
44            ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Oui    Non  Non   
46            ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Oui    Non  Non   
49            ABIDJAN  AUTONOME D'ABIDJAN  ...   Oui    Non   Non    Oui  Non   
...               ...                 ...  ...   ...    ...   ...    ...  ...   
64240  CENTRE (rural)             MORONOU  ...   Oui    Non   Non    Oui  Non   
64254  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Oui  Non   
64289  CENTRE (rural)             MORONOU  ...   Non    Non   Non    Non  Non   
64315  CENTRE (rural)             MORONOU  ...   Non    Oui   Non    Oui  Non   
64316  CENTRE (rural)             MORONOU  ...   Non    Oui   Non    Oui  Non   

             superf  bancarise volhor_manquant salaire_mois_manquant  \
0      0.000000e+00          1               0                     0   
40     0.000000e+00          1               0                     0   
44     0.000000e+00          1               0                     0   
46     0.000000e+00          1               0                     0   
49     0.000000e+00          1               0                     0   
...             ...        ...             ...                   ...   
64240  0.000000e+00          1               0                     0   
64254  0.000000e+00          1               0                     0   
64289  4.577763e+00          1               0                     0   
64315  1.650028e+07          1               0                     1   
64316  1.650028e+07          1               0                     0   

      Stg_formel  
0              1  
40             1  
44             1  
46             1  
49             1  
...          ...  
64240          1  
64254          1  
64289          1  
64315          1  
64316          1  

[1979 rows x 81 columns]

## Variables à laisser

In [118]:
df_stat = df_anstat.drop(['idmen', 'idind', 'country', 'year', 'PanelHH', 'numind', 'vague', 'moustiq', 'zaemil', 'has_account_bank', 'est_scolarise'], axis=1)
df_stat

agro_zone         region_name department        city milieu_resid  \
0       ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN     ABIDJAN       Urbain   
1       ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN     ABIDJAN       Urbain   
2       ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN     ABIDJAN       Urbain   
3       ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN     ABIDJAN       Urbain   
4       ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN     ABIDJAN       Urbain   
...         ...                 ...        ...         ...          ...   
64486    CENTRE             MORONOU    M'BATTO  TIEMELEKRO        Rural   
64487    CENTRE             MORONOU    M'BATTO  TIEMELEKRO        Rural   
64488    CENTRE             MORONOU    M'BATTO  TIEMELEKRO        Rural   
64489    CENTRE             MORONOU    M'BATTO  TIEMELEKRO        Rural   
64490    CENTRE             MORONOU    M'BATTO  TIEMELEKRO        Rural   

        hhweight resid       sex  age_num            lien  ... frigo cuisin  \
0      1098.1172   Oui   Féminin       29     Fils, Fille  ...   Oui    Non   
1      1098.1172   Oui   Féminin       17     Fils, Fille  ...   Oui    Non   
2      1098.1172   Oui  Masculin       15     Fils, Fille  ...   Oui    Non   
3      1098.1172   Oui   Féminin       12     Fils, Fille  ...   Oui    Non   
4      1098.1172   Oui   Féminin       34  Conjoint ( e )  ...   Oui    Non   
...          ...   ...       ...      ...             ...  ...   ...    ...   
64486   242.7796   Oui  Masculin       11     Neveu/Nièce  ...   Non    Non   
64487   242.7796   Oui  Masculin        7     Fils, Fille  ...   Non    Non   
64488   242.7796   Oui  Masculin       10     Neveu/Nièce  ...   Non    Non   
64489   242.7796   Oui  Masculin        4     Fils, Fille  ...   Non    Non   
64490   242.7796   Oui  Masculin       36  Chef de ménage  ...   Non    Non   

      ordin decod  car     superf bancarise volhor_manquant  \
0       Non   Oui  Non        0.0         1               0   
1       Non   Oui  Non        0.0         0               1   
2       Non   Oui  Non        0.0         0               1   
3       Non   Oui  Non        0.0         0               1   
4       Non   Oui  Non        0.0         0               1   
...     ...   ...  ...        ...       ...             ...   
64486   Non   Oui  Non  8566398.0         0               1   
64487   Non   Oui  Non  8566398.0         0               1   
64488   Non   Oui  Non  8566398.0         0               1   
64489   Non   Oui  Non  8566398.0         0               1   
64490   Non   Oui  Non  8566398.0         1               0   

      salaire_mois_manquant Stg_formel  
0                         0          1  
1                         1          0  
2                         1          0  
3                         1          0  
4                         1          0  
...                     ...        ...  
64486                     1          0  
64487                     1          0  
64488                     1          0  
64489                     1          0  
64490                     1          0  

[64474 rows x 70 columns]

In [119]:
df_fin = df_anstat.drop(['year', 'PanelHH', 'numind', 'vague', 'moustiq', 'zaemil', 'has_account_bank', 'est_scolarise'], axis=1)
df_fin

idmen  idind country agro_zone         region_name department  \
0          2      4     CIV   ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN   
1          2      5     CIV   ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN   
2          2      3     CIV   ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN   
3          2      6     CIV   ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN   
4          2      2     CIV   ABIDJAN  AUTONOME D'ABIDJAN    ABIDJAN   
...      ...    ...     ...       ...                 ...        ...   
64486  13863  64490     CIV    CENTRE             MORONOU    M'BATTO   
64487  13863  64489     CIV    CENTRE             MORONOU    M'BATTO   
64488  13863  64491     CIV    CENTRE             MORONOU    M'BATTO   
64489  13863  64488     CIV    CENTRE             MORONOU    M'BATTO   
64490  13863  64486     CIV    CENTRE             MORONOU    M'BATTO   

             city milieu_resid   hhweight resid  ... frigo  cuisin ordin  \
0         ABIDJAN       Urbain  1098.1172   Oui  ...   Oui     Non   Non   
1         ABIDJAN       Urbain  1098.1172   Oui  ...   Oui     Non   Non   
2         ABIDJAN       Urbain  1098.1172   Oui  ...   Oui     Non   Non   
3         ABIDJAN       Urbain  1098.1172   Oui  ...   Oui     Non   Non   
4         ABIDJAN       Urbain  1098.1172   Oui  ...   Oui     Non   Non   
...           ...          ...        ...   ...  ...   ...     ...   ...   
64486  TIEMELEKRO        Rural   242.7796   Oui  ...   Non     Non   Non   
64487  TIEMELEKRO        Rural   242.7796   Oui  ...   Non     Non   Non   
64488  TIEMELEKRO        Rural   242.7796   Oui  ...   Non     Non   Non   
64489  TIEMELEKRO        Rural   242.7796   Oui  ...   Non     Non   Non   
64490  TIEMELEKRO        Rural   242.7796   Oui  ...   Non     Non   Non   

      decod  car     superf bancarise  volhor_manquant salaire_mois_manquant  \
0       Oui  Non        0.0         1                0                     0   
1       Oui  Non        0.0         0                1                     1   
2       Oui  Non        0.0         0                1                     1   
3       Oui  Non        0.0         0                1                     1   
4       Oui  Non        0.0         0                1                     1   
...     ...  ...        ...       ...              ...                   ...   
64486   Oui  Non  8566398.0         0                1                     1   
64487   Oui  Non  8566398.0         0                1                     1   
64488   Oui  Non  8566398.0         0                1                     1   
64489   Oui  Non  8566398.0         0                1                     1   
64490   Oui  Non  8566398.0         1                0                     1   

      Stg_formel  
0              1  
1              0  
2              0  
3              0  
4              0  
...          ...  
64486          0  
64487          0  
64488          0  
64489          0  
64490          0  

[64474 rows x 73 columns]

# **Exportation Clean_Datas**

In [120]:
# Pour les statistiques

df_stat.to_csv("../DATAS/ANSTAT2021_dataset_Clean.csv", index=False, encoding="utf-8")

In [121]:
# Pour les statistiques

df_fin.to_csv("../DATAS/ANSTAT2021_dataset_AllVars_CLEAN.csv", index=False, encoding="utf-8")

In [122]:
df_fin.shape

(64474, 73)

In [123]:
df_fin.columns

Index(['idmen', 'idind', 'country', 'agro_zone', 'region_name', 'department',
       'city', 'milieu_resid', 'hhweight', 'resid', 'sex', 'age_num', 'lien',
       'marital_status', 'religion', 'ethnie', 'nation', 'agemar', 'mal30j',
       'aff30j', 'arrmal', 'durarr', 'con30j', 'hos12m', 'couvmal', 'handit',
       'handig', 'alfa', 'alfa2', 'scol', 'educ_scol', 'educ_hi', 'diplome',
       'telpor', 'internet', 'activ7j', 'activ12m', 'branch', 'sectins', 'csp',
       'volhor', 'salaire', 'emploi_sec', 'sectins_sec', 'csp_sec',
       'volhor_sec', 'salaire_sec', 'serviceconsult', 'persconsult',
       'salaire_mois', 'salaire_sec_mois', 'rev_total_mois', 'log_revenu',
       'sans_revenu', 'age_grp', 'a_assurance', 'alphabete', 'logem',
       'elec_ac', 'elec_ur', 'elec_ua', 'tv', 'fer', 'frigo', 'cuisin',
       'ordin', 'decod', 'car', 'superf', 'bancarise', 'volhor_manquant',
       'salaire_mois_manquant', 'Stg_formel'],
      dtype='object')

In [124]:
df_stat.shape

(64474, 70)

In [125]:
df_stat.columns

Index(['agro_zone', 'region_name', 'department', 'city', 'milieu_resid',
       'hhweight', 'resid', 'sex', 'age_num', 'lien', 'marital_status',
       'religion', 'ethnie', 'nation', 'agemar', 'mal30j', 'aff30j', 'arrmal',
       'durarr', 'con30j', 'hos12m', 'couvmal', 'handit', 'handig', 'alfa',
       'alfa2', 'scol', 'educ_scol', 'educ_hi', 'diplome', 'telpor',
       'internet', 'activ7j', 'activ12m', 'branch', 'sectins', 'csp', 'volhor',
       'salaire', 'emploi_sec', 'sectins_sec', 'csp_sec', 'volhor_sec',
       'salaire_sec', 'serviceconsult', 'persconsult', 'salaire_mois',
       'salaire_sec_mois', 'rev_total_mois', 'log_revenu', 'sans_revenu',
       'age_grp', 'a_assurance', 'alphabete', 'logem', 'elec_ac', 'elec_ur',
       'elec_ua', 'tv', 'fer', 'frigo', 'cuisin', 'ordin', 'decod', 'car',
       'superf', 'bancarise', 'volhor_manquant', 'salaire_mois_manquant',
       'Stg_formel'],
      dtype='object')

# **RESUME**